File: app/src/main/java/pl/gocards/ui/decks/xml/standard/DeckViewHolder.java
package pl.gocards.ui.decks.xml.standard;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;

import pl.gocards.R;
import pl.gocards.ui.base.recyclerview.BaseViewHolder;
import pl.gocards.ui.decks.xml.standard.dialog.NoCardsToRepeatDialog;

/**
 * D_R_02 Show all decks
 * @author Grzegorz Ziemski
 */
public class DeckViewHolder extends BaseViewHolder implements View.OnClickListener {

    protected boolean noCardsToRepeat;

    public DeckViewHolder(@NonNull View itemView, @NonNull DeckViewAdapter adapter) {
        super(itemView, adapter);
        initMoreImageView();
        itemView.setOnClickListener(this);
    }

    private void initMoreImageView() {
        View moreTextView = getMoreImageView();
        moreTextView.setOnClickListener(this::showPopupMenu);
    }

    protected void showPopupMenu(View moreTextView) {
        new DeckPopupMenu(
                getAdapter(),
                getBindingAdapterPosition(),
                moreTextView
        ).showPopupMenu();
    }

    @Override
    public void onClick(View view) {
        if (noCardsToRepeat) {
            showNoCardsToRepeatDialog();
        } else {
            getAdapter().onItemClick(getBindingAdapterPosition());
        }
    }

    /**
     * C_R_31 No more cards to repeat
     */
    protected void showNoCardsToRepeatDialog() {
        NoCardsToRepeatDialog dialog = new NoCardsToRepeatDialog();
        dialog.show(requireActivity().getSupportFragmentManager(), "NoCardsToRepeatDialog");
    }

    /* -----------------------------------------------------------------------------------------
     * Gets/Sets for View
     * ----------------------------------------------------------------------------------------- */

    @NonNull
    protected TextView getNameTextView() {
        return itemView.findViewById(R.id.nameTextView);
    }

    @NonNull
    protected TextView getTotalTextView() {
        return itemView.findViewById(R.id.totalTextView);
    }

    @NonNull
    protected TextView getNewTextView() {
        return itemView.findViewById(R.id.newTextView);
    }

    @NonNull
    protected TextView getRevTextView() {
        return itemView.findViewById(R.id.revTextView);
    }

    @NonNull
    protected View getMoreImageView() {
        return itemView.findViewById(R.id.moreImageView);
    }

    /* -----------------------------------------------------------------------------------------
     * Gets/Sets
     * ----------------------------------------------------------------------------------------- */

    @NonNull
    public DeckViewAdapter getAdapter() {
        return (DeckViewAdapter) super.getAdapter();
    }

    public void setNoCardsToRepeat(boolean noCardsToRepeat) {
        this.noCardsToRepeat = noCardsToRepeat;
    }
}

File: app/src/main/java/pl/gocards/ui/decks/decks/view/DeckViewHolder.kt
package pl.gocards.ui.decks.decks.view

import android.app.Activity
import android.view.View
import android.widget.TextView
import androidx.compose.runtime.Composable
import androidx.recyclerview.widget.RecyclerView
import pl.gocards.R
import pl.gocards.databinding.ItemDeckBinding
import pl.gocards.ui.common.popup.menu.ShowPopupMenuAtPos
import pl.gocards.ui.decks.decks.ListDecksAdapter
import pl.gocards.ui.decks.recent.ListRecentDecksAdapter

/**
 * D_R_02 Show all decks
 * @author Grzegorz Ziemski
 */
class DeckViewHolder(
    binding: ItemDeckBinding,
    val activity: Activity,
    val adapter: ListDecksAdapter
) : RecyclerView.ViewHolder(binding.root), View.OnClickListener {
    var noCardsToRepeat = true

    init {
        initMoreButton()
        binding.root.setOnClickListener(this)
    }

    private fun initMoreButton() {
        val moreTextView = getMoreImageView()
        moreTextView.setOnClickListener { view: View ->
            ShowPopupMenuAtPos(activity).createPopupMenu(view)
            { onDismiss -> CreatePopupMenu(onDismiss) }
        }
    }

    @Composable
    private fun CreatePopupMenu(onDismiss: () -> Unit) {
        DeckPopupMenu(
            onDismiss = onDismiss,
            onClickBrowseCards = if (adapter.isPremium) {
                { adapter.newBrowseCardsActivity(bindingAdapterPosition) }
            } else null,
            onClickListCards = { adapter.newListCardsActivity(bindingAdapterPosition) },
            onClickNewCard = { adapter.newNewCardActivity(bindingAdapterPosition) },
            onClickCutDeck = if (adapter is ListRecentDecksAdapter) null else {
                { adapter.cutDeck(bindingAdapterPosition) }
            },
            onClickRenameDeck = { adapter.showRenameDeckDialog(bindingAdapterPosition) },
            onClickDeleteDeck = { adapter.showDeleteDeckDialog(bindingAdapterPosition) },
            onClickShowMenuBottom = { adapter.showMoreDeckMenu(bindingAdapterPosition) }
        )
    }

    override fun onClick(view: View) {
        if (noCardsToRepeat) {
            adapter.showNoCardsToRepeatDialog()
        } else {
            adapter.onItemClick(getBindingAdapterPosition())
        }
    }

    /* -----------------------------------------------------------------------------------------
     * Gets/Sets for View
     * ----------------------------------------------------------------------------------------- */

    fun getNameTextView(): TextView {
        return itemView.findViewById(R.id.nameTextView)
    }

    fun getTotalTextView(): TextView {
        return itemView.findViewById(R.id.totalTextView)
    }

    fun getNewTextView(): TextView {
        return itemView.findViewById(R.id.newTextView)
    }

    fun getRevTextView(): TextView {
        return itemView.findViewById(R.id.revTextView)
    }

    private fun getMoreImageView(): View {
        return itemView.findViewById(R.id.moreImageView)
    }
}